x-pack/test/functional/services/ml/deploy_models_flyout.ts

```diff
@@ -0,0 +1,296 @@
+      await testSubjects.existOrFail('mlTrainedModelsInferenceAdvancedSettingsConditionTextArea');
+      await testSubjects.existOrFail('mlTrainedModelsInferenceAdvancedSettingsTagInput');
+    }
+
+    public async completeTrainedModelsInferenceFlyoutPipelineConfig(
+      expectedValues: {
+        inferenceConfig: any;
+        editedInferenceConfig?: any;
+        fieldMap: any;
+        editedFieldMap?: any;
+      },
+      editDefaults: boolean = false
+    ) {
+      const { inferenceConfig, editedInferenceConfig, fieldMap, editedFieldMap } = expectedValues;
+      // Check all defaults
+      await this.assertTrainedModelsInferenceFlyoutPipelineConfigValues(inferenceConfig, fieldMap);
+
+      if (editDefaults) {
+        await this.setTrainedModelsInferenceFlyoutCustomPipelineConfig({
+          condition: "ctx?.network?.name == 'Guest'",
+          editedFieldMap,
+          editedInferenceConfig,
+          tag: 'tag',
+        });
+      }
+
+      await this.deployModelsContinue('mlTrainedModelsInferenceOnFailureStep');
+    }
+
+    public async completeTrainedModelsInferenceFlyoutOnFailure(
+      expectedOnFailure: IngestInferenceProcessor['on_failure'],
+      editDefaults: boolean = false
+    ) {
+      await retry.tryForTime(30 * 1000, async () => {
+        // Switch should default to unchecked
+        const ignoreFailureSelected =
+          (await testSubjects.getAttribute(
+            'mlTrainedModelsInferenceIgnoreFailureSwitch',
+            'aria-checked'
+          )) === 'true';
+        expect(ignoreFailureSelected).to.eql(false);
+        // Switch should default to checked
+        const takeActionOnFailureSelected =
+          (await testSubjects.getAttribute(
+            'mlTrainedModelsInferenceTakeActionOnFailureSwitch',
+            'aria-checked'
+          )) === 'true';
+        expect(takeActionOnFailureSelected).to.eql(true);
+      });
+
+      const defaultOnFailure = await testSubjects.getVisibleText(
+        'mlTrainedModelsInferenceOnFailureCodeBlock'
+      );
+      expect(JSON.parse(defaultOnFailure)).to.eql(expectedOnFailure);
+
+      if (editDefaults) {
+        await retry.tryForTime(30 * 1000, async () => {
+          // switch ignore failure to true
+          await testSubjects.click('mlTrainedModelsInferenceIgnoreFailureSwitch');
+          // Switch should now be checked
+          const isIgnoreFailureSelected =
+            (await testSubjects.getAttribute(
+              'mlTrainedModelsInferenceIgnoreFailureSwitch',
+              'aria-checked'
+            )) === 'true';
+          expect(isIgnoreFailureSelected).to.eql(true);
+        });
+      }
+      await this.deployModelsContinue('mlTrainedModelsInferenceTestStep');
+      // skip test step
+      await this.deployModelsContinue('mlTrainedModelsInferenceReviewAndCreateStep');
+    }
+
+    public async completeTrainedModelsInferenceFlyoutCreateStep(expectedConfig: any) {
```

Comment:
type is missing

Reply:
Added in 001a8301d6190c0a994c65cff3a18105aeda53b4